lib/findtoken.h

```diff
@@ -92,7 +96,8 @@ bool findTokensSkipDeadCodeImpl(const Library& library,
             if (found(tok))
                 return true;
         }
-        if (Token::Match(tok, "if|for|while (") && Token::simpleMatch(tok->linkAt(1), ") {")) {
+        //if (Token::Match(tok, "if|for|while (") && Token::simpleMatch(tok->linkAt(1), ") {")) {
+        if (findTokensSkipDeadCodeImplMatch_1(tok)) {
```

Comment:
Why cant the match compiler be updated to automatically generate this? This seems very problematic.

Reply:
Seems obvious in hindsight. But thinking about it I think there might be problems.

That would cause the `matchcompiler.h` include to spill which seems undesired. Same with the match functions. The latter will probably cause conflicts in other matchcompiled files which include this.

Can check thought if if it would work if we also process header files. Would be a better approach - albeit I would prefer if it only were temporary.

Reply:
> That would cause the matchcompiler.h include to spill which seems undesired

What do you mean spill? There is a header guard.

> Same with the match functions. The latter will probably cause conflicts in other matchcompiled files which include this.

I thought each function is appended a unique number.

Reply:
> > That would cause the matchcompiler.h include to spill which seems undesired
> 
> What do you mean spill? There is a header guard.

I mean that it will be includes in all files including this even when not necessary. It is just not clean.

> > Same with the match functions. The latter will probably cause conflicts in other matchcompiled files which include this.
> 
> I thought each function is appended a unique number.

I am not sure if that is unique globally or locally. The first is unnecessary since we only generate functions within source files. But will immediately come up if I try to include the headers.

Reply:
Okay. I just remembered the main issue.

The files need to be matchcompiled thus you need to store the modified files and build those. But if you do that with headers you also need to inject an additional header path in release builds and you two headers of the same name which might also cause issues.

So it is not just include the `*.h` in the matchcompiler.

Reply:
We have no "other directories". We are using a flat structure and introducing an additional folder (for a single file mind you) will have a ripple effect. This needs to be addressed in a different way for each build system and I have the feeling that would not work at all for the Visual Studio release build we perform in the CI.

I tried introducing support for subfolders for the sources in the matchcompiler at some point. It was a pain, very hacky and I never got it to work fully so I abandoned it. I did that when I proposed the splitting of `valueflow.cpp`.

On a side note. I just realized that the matchcompiler is not integrated at all in the Visual Studio project file we are providing so people using that to build their local binary will have a much slower executable. Another reason to dump these files and go with CMake all the way.

Reply:
> We have no "other directories". We are using a flat structure and introducing an additional folder (for a single file mind you) will have a ripple effect. This needs to be addressed in a different way for each build system and I have the feeling that would not work at all for the Visual Studio release build we perform in the CI.

We can always create a directory. Then we can get it running using cmake. For the other build systems, its just a matter of adding an include path. There wont be a perf regression for these other build systems because they are not using the matchcompiler on header files in the first place. Users of these build systems can provide the support to improve it. We shouldn't make it a barrier for improving code quality.

Also, it is a single file, but this will make it easier to add more header-only files in the future. 

> Another reason to dump these files and go with CMake all the way.

I agree. CMake already has a lot of enhancements already over the other builds such as parallel test running, and this will be another additional enhancement.